src/commands/synthetics/tunnel/tunnel.ts

```diff
@@ -30,24 +28,30 @@ export interface TunnelInfo {
   privateKey: string
 }
 
+export interface TunnelReporter {
+  error(message: string): void
+  log(message: string): void
+  warn(message: string): void
+}
+
 export class Tunnel {
-  private connected = false
-  private forwardedSockets: Set<Socket> = new Set()
   private FORWARDING_TIMEOUT = 40000 as const
-  private log: (message: string) => void
-  private logError: (message: string) => void
-  private logWarning: (message: string) => void
-  private multiplexer?: Multiplexer
+
+  private sshConfig: ServerConfig
   private privateKey: string
   private publicKey: ParsedKey
-  private sshConfig: ServerConfig
-  private ws: WebSocket
 
-  constructor(private url: string, private testIDs: string[], proxy: ProxyConfiguration, reporter: MainReporter) {
-    this.log = (message: string) => reporter.log(`[${chalk.bold.blue('Tunnel')}] ${message}\n`)
-    this.logError = (message: string) => reporter.error(`[${chalk.bold.red('Tunnel')}] ${message}\n`)
-    this.logWarning = (message: string) => reporter.log(`[${chalk.bold.yellow('Tunnel')}] ${message}\n`)
+  private connected = false
+  private ws: WebSocket
+  private multiplexer?: Multiplexer
+  private forwardedSockets: Set<Socket> = new Set()
 
+  constructor(
+    private url: string,
+    private testIDs: string[],
+    proxyAgent?: ReturnType<typeof ProxyAgent>,
+    private reporter?: TunnelReporter
```

Comment:
Why make the reporter optional ?

Reply:
I was hesitant making it optional, some integrations may not need it but maybe we can instead make its methods optional 🤔

Reply:
Ah gotcha, this was with integrations in mind. Then yeah, from a core functionality standpoint the reporter is not necessarily "required".